packages/mangrove/src/kandel.ts

```diff
@@ -0,0 +1,88 @@
+import { Address, Amount, TxRef } from "@proximaone/stream-schema-base";
+
+
+export type SeederEvent = (NewAaveKandel | NewKandel) & {
+  tx: TxRef;
+  id: string;
+  chainId: number;
+  address: Address;
+};
+
+export type KandelEvent = (
+  | AllBids
+  | AllAsks
+  | SetParams
+  | Credit
+  | Debit
+  | SetGeometricParams
+  | SetLength
+) & {
+  tx: TxRef;
+  id: string;
+  chainId: number;
+  address: Address;
+};
+
+export interface NewAaveKandel {
+  type: "NewAaveKandel";
+
+  owner: Address;
+  base: Address;
+  quote: Address;
+  aaveKandel: Address;
+  reserveId: Address;
+}
+
+export interface NewKandel {
+  type: "NewKandel";
+
+  owner: Address;
+  base: Address;
+  quote: Address;
+  kandel: Address;
+}
+
+export interface AllBids {
+  type: "AllBids";
+}
+
+export interface AllAsks {
+  type: "AllAsks";
+}
+
+export interface SetParams {
+  type: "SetParams";
+  compoundRates?: {
+    base?: number;
+    quote?: number;
```

Comment:
If it's possible that only one of them is present, let's move them out of `compoundRates` struct. I proposed that form to guarantee that either they are both present or none of them.